tests/test_crm.py

```diff
@@ -0,0 +1,538 @@
+    avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_ipv4_nexthop_available')
+
+    # add nexthop and update available counter
+    dvs.servers[0].runcmd("ping -c 1 10.0.0.0")
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_IPV4_NEXTHOP_ENTRY', '999')
+
+    time.sleep(60)
+
+    # get counters
+    new_used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_ipv4_nexthop_used')
+    new_avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_ipv4_nexthop_available')
+
+    assert new_used_counter - used_counter == 1
+    assert avail_counter - new_avail_counter == 1
+
+    # remove nexthop and update available counter
+    dvs.runcmd("ip -s -s neigh flush 10.0.0.1")
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_IPV4_NEXTHOP_ENTRY', '1000')
+
+    time.sleep(60)
+
+    # get counters
+    new_used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_ipv4_nexthop_used')
+    new_avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_ipv4_nexthop_available')
+
+    assert new_used_counter == used_counter
+    assert new_avail_counter == avail_counter
+
+
+def test_CrmIpv6Nexthop(dvs):
+
+    dvs.runcmd("ifconfig Ethernet0 inet6 add fc00::1/126 up")
+
+    dvs.servers[0].runcmd("ifconfig eth0 inet6 add fc00::2/126")
+    dvs.servers[0].runcmd("ip -6 route add default via fc00::1")
+
+    dvs.runcmd("crm config polling interval 1")
+
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_IPV6_NEXTHOP_ENTRY', '1000')
+
+    time.sleep(5*60)
+
+    # get counters
+    used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_ipv6_nexthop_used')
+    avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_ipv6_nexthop_available')
+
+    # add nexthop and update available counter
+    dvs.servers[0].runcmd("ping6 -c 1 fc00::1")
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_IPV6_NEXTHOP_ENTRY', '999')
+
+    time.sleep(60)
+
+    # get counters
+    new_used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_ipv6_nexthop_used')
+    new_avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_ipv6_nexthop_available')
+
+    assert new_used_counter - used_counter == 1
+    assert avail_counter - new_avail_counter == 1
+
+    # remove nexthop and update available counter
+    dvs.runcmd("ip -s -s neigh flush fc00::2")
+    setReadOnlyAttr(dvs, 'SAI_OBJECT_TYPE_SWITCH', 'SAI_SWITCH_ATTR_AVAILABLE_IPV6_NEXTHOP_ENTRY', '1000')
+
+    time.sleep(60)
+
+    # get counters
+    new_used_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_ipv6_nexthop_used')
+    new_avail_counter = getCrmCounterValue(dvs, 'STATS', 'crm_stats_ipv6_nexthop_available')
+
+    assert new_used_counter == used_counter
+    assert new_avail_counter == avail_counter
+
+
+def test_CrmIpv4Neighbor(dvs):
+
+    dvs.runcmd("ifconfig Ethernet0 10.0.0.0/31 up")
```

Comment:
>dvs.runcmd("ifconfig Ethernet0 10.0.0.0/31 up") [](start = 4, length = 47)

a simpler approach is to add static arp, then you do not need to setup server side.